### PR TITLE
[SPARK-39035][SQL][TESTS] Add tests for options from `to_csv` and `from_csv`.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fill up the missing test for options from CSV read/write functions.

### Why are the changes needed?

For better testability and correct the documents.

### Does this PR introduce _any_ user-facing change?

No, it's dev-only.

### How was this patch tested?

Added some UTs for CSV read/write options.

### CODE REVIEW

**Review by HyukjinKwon — DISMISSED** [review]
~~LGTM if tests pass.~~ cc @tgravescs FYI

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Sorry, the tests failed https://github.com/apache/spark/runs/6226791783. @itholic can you run the tests, and open a PR please?

**Comment by HyukjinKwon** [comment]
I reverted this.

File: sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala [hunk 1]
```diff
@@ -79,6 +79,108 @@ class CsvFunctionsSuite extends QueryTest with SharedSparkSession {
...
+      df.select(from_csv($"value", schema, options)),
+      Row(Row(java.sql.Date.valueOf("2015-08-26"))))
+  }
+
+  test("from_csv with option (maxCharsPerColumn)") {
+    val df = Seq("12345").toDS()
+    val schema = new StructType().add("str", StringType)
+    val options = Map("maxCharsPerColumn" -> "2")
+
+    val exception = intercept[SparkException] {
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala:177]: Can we also test the root cause exception? e.g., `TextParsingException`. You can get it via `e.getCause`.

**Review by HyukjinKwon — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**